modules/stochastic_tools/src/userobjects/NeuralNetworkUserObject.C

```diff
@@ -0,0 +1,186 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NeuralNetworkUserObject.h"
+#include "MooseEnum.h"
+#include "pugixml.h"
+
+registerMooseObject("StochasticToolsApp", NeuralNetworkUserObject);
+
+template <>
+InputParameters
+validParams<NeuralNetworkUserObject>()
+{
+  InputParameters params = GeneralUserObject::validParams();
+  params.addClassDescription("Reconstructs a neural network from a file and evaluates it");
+  params.addRequiredParam<FileName>("weights_file", "Name of the file with the neuron weights");
+  return params;
+}
+
+NeuralNetworkUserObject::NeuralNetworkUserObject(const InputParameters & parameters)
+  : GeneralUserObject(parameters),
+    _weightsFile(getParam<FileName>("weights_file")),
+    _layerActivationFunctionEnum(MultiMooseEnum("SIGMOID SOFTSIGN TANH LINEAR LOGSIGMOID"))
```

Comment:
This should be an `std::vector` of an `enum class`. MultiMooseEnum is for capturing input parameter values.